src/main/scala/com/gu/invoicing/preview/Impl.scala

```diff
@@ -22,6 +22,17 @@ object Impl {
       .accountId
   }
 
+  /* Needed to obtain chargeAmount with tax included as billing-preview is not inclusive of tax */
+  def getRatePlanCharges(subscriptionName: String, startDate: LocalDate) =
+    Http(s"$zuoraApiHost/v1/subscriptions/$subscriptionName?specific-segment&as-of-date=$startDate")
+      .header("Authorization", s"Bearer $accessToken")
+      .asString
+      .body
+      .pipe(read[Subscription](_))
+      .ratePlans
+      .flatMap(_.ratePlanCharges)
+      .filterNot(_.price < 0.0)
```

Comment:
Perhaps `.filter(_.price > 0.0)`? 
as a free rate plan charge would be irrelevant, unless it's useful.